### ISSUE TITLE
LR minimal value

### ISSUE THREAD

**Issue by PjuralSzymon** [PjuralSzymon/growingNN_pytorch#72]
we had an issue that lr often is almost allways minimal alue:

<img width="900" height="600" alt="Image" src="https://github.com/user-attachments/assets/7c4f391b-dd3c-4169-bba1-47c93252a3e7" />

<html>
<body>
<pre data-pm-slice="1 1 []"><code class="language-text">effective_lr = max(0.001, global_lr * recovery_factor)</code></pre><p>Logistic <code>k=10</code> keeps <code>recovery_factor</code> small in the first half of warmup. The product stays under 0.001 until the factor is large enough, so the floor <strong>erases the early ramp</strong>.</p><h2>Why the floor lasts longer each generation</h2><p>Exponential base keeps falling: <code>0.01 * 0.98^t</code>. The factor needed to beat the floor is <code>0.001 / global_lr</code>. That threshold grows as <code>global_lr</code> approaches 0.001.</p>
After action at epoch | Approx global LR | Factor needed to leave 0.001 | Floor lasts about
-- | -- | -- | --
20 | 0.0068 | 0.15 | 4 epochs
60 | 0.0030 | 0.33 | 5–6 epochs
90 | 0.0016 | 0.62 | 6–7 epochs, tiny hump

<p>By the last generation the peak of the hump is only ~0.0016, so the plot looks flat at the minimum for almost the whole generation. This is the floor eating a decaying base, not a counter that fails to advance.</p><p>A second issue: <code>[WarmupSchedule.alpha_scheduler](growingnn/training/lr_scheduler_action.py)</code> clamps the <strong>0–1 recovery factor</strong> with <code>MIN_LEARNING_RATE</code> as if it were an absolute LR. Compose should use the raw factor; only the final LR should respect the floor.</p><p>Do <strong>not</strong> lower <code>MIN_LEARNING_RATE</code>. The skill keeps that floor unless the task changes it.</p>
</body>
</html>

**Comment by PjuralSzymon** [comment]
Issue created during working on: https://github.com/PjuralSzymon/growingNN_pytorch/pull/75
